### PR TITLE
plot: make tick labels non-ambiguous

### LINKED ISSUE #301: Tick label are indisintinguishable for small differences
Current precision requirement for tick values allows resolving min and max tick values, but does not provide ability to resolve differences between adjacent tick values when when they are close spaced.  Note all tick labels on colorbar heatmap exactly equal at 0.0002

tickprecisionambiguity

### PR DESCRIPTION
**Pull request by kortschak**
Fixes #301.

@jasonpfox @sbinet @eaburns Please take a look.

### CODE REVIEW

**Comment by jasonpfox** [comment]
Looks great, thanks.

File: axis.go [hunk 1]
```diff
@@ -384,7 +384,7 @@ func (DefaultTicks) Ticks(min, max float64) (ticks []Tick) {
 	}
 	majorDelta := float64(majorMult) * tens
 	val := math.Floor(min/majorDelta) * majorDelta
-	prec := maxInt(precisionOf(min), precisionOf(max))
+	prec := precisionOf(majorDelta)
```
Comment (btracey) [inline axis.go]: And the logic is simpler!

**Comment by btracey** [comment]
LGTM

File: axis.go (hunk 1 above)
Comment (kortschak) [inline axis.go]: Yes, thanks to @jasonpfox for that observation. I don't really know what I was thinking when I wrote the original.

File: axis_test.go [hunk 2]
```diff
@@ -35,19 +35,25 @@ func TestAxisSmallTick(t *testing.T) {
...
+			Max:    0.00015,
+			Labels: []string{"0.0001", "0.00011", "0.00012", "0.00013", "0.00014"},
+		},
 	} {
 		ticks := d.Ticks(test.Min, test.Max)
 		var count int
 		for _, tick := range ticks {
 			if tick.Label != "" {
+				if count >= len(test.Labels) {
+					t.Errorf("Too many tick labels")
```
Comment (sbinet) [inline axis_test.go]: please provide a "got=%d want=%d" information text so we know which test failed.

**Comment by eaburns** [comment]
Sorry that I'm late, LGTM.

**Pull request merged**